netlify/netlify-plugin-gatsby
# feat: Optionally load gatsby datastore in lambdas

### Summary

Introduce an environment variable that controls whether the Gatsby datastore is bundled in with the lambdas or is uploaded to the CDN and downloaded on first load of the lambdas.

If the environment variable (`LOAD_GATSBY_LMDB_DATASTORE_FROM_CDN`) is enabled, the plugin now also sends a pre-warm request to the SSR, DSG, and API functions. This is because it can take some time to initially download the datastore, and we want to mitigate the impact on our user's end-users of this operation (they could be waiting an extended period of time and it might look like the page is hanging).

**Additional changes**
* Introduce Jest configuration, and the `chance`, `uuid`, `tmp-package`, and `tmp-promise` packages for unit tests
* Increased minimum node version to `14.17.0` due to the use of the AbortController in the `onSuccess` build step
* Moved `execa` to a `devDependency` from `peerDependency` within the `plugin/package.json`. This is because we're using it directly within a test, and we also want to ensure that it's on v5 as v6 is entirely in pure ESM which was causing build issues within the project (see [release notes](https://github.com/sindresorhus/execa/releases/tag/v6.0.0))

### Test plan

1. Visit the Deploy Preview  - https://ep-optionally-load-gatsby-datastore-in-lambdas.netlify.app/
2. Click on 'Blog' > 'Second Post'. This is a DSG page so the page should be available and render properly when clicked on.

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal
Partially addresses: https://github.com/netlify/netlify-plugin-gatsby/issues/320

Doesn't include the following from the linked issue:
* setting headers config to add basic auth to the db file too
* modifying the datastore independently to the rest of the build

![signal-2022-04-21-20-37-31-546](https://user-images.githubusercontent.com/5655473/168294018-2f1452f1-69f6-4d28-add4-6e6cd22c5a9e.jpg)


### Standard checks:




---

🧪 Once merged, make sure to update the version if needed and that it was
published correctly.
